src/Application/Models/Broker.js

```diff
@@ -42,39 +41,40 @@ class Broker {
     return this._assets.filter((asset) => asset.unlocked);
   }
 
-  makeBuy(assetId) {
+  // TODO: Change this to find index where
+  buyAsset(assetId) {
     const asset = _.findWhere(this._assets, { id: assetId });
-    const newAsset = asset.makeBuy(PRICE_INCREASE_PERCENTAGE);
+    const newAsset = asset.buy();
 
     const arrayCopy = this._assets.slice();
-    arrayCopy[asset.id] = newAsset;
+    arrayCopy[asset.id - 1] = newAsset;
     return new Broker(arrayCopy);
   }
```

Comment:
Should make this actually replace the exactly element rather than assume it is in order of ID.